File: evaluation/microservice-mongodb-springdata-java/src/main/java/cz/cuni/mff/java/microservice/model/embedded/OrdersEWithCustomerWithNationWithRegion.java
package cz.cuni.mff.java.microservice.model.embedded;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;

@Document(collection = "ordersEWithCustomerWithNationWithRegion")
public class OrdersEWithCustomerWithNationWithRegion {
    @Id
    private int o_orderkey;
    private LocalDate o_orderdate;
    private CustomerEOnlyCNameCNation o_customer;

    public OrdersEWithCustomerWithNationWithRegion() {}

    public OrdersEWithCustomerWithNationWithRegion(int o_orderkey, LocalDate o_orderdate,
                                                   CustomerEOnlyCNameCNation o_customer) {
        this.o_orderkey = o_orderkey;
        this.o_orderdate = o_orderdate;
        this.o_customer = o_customer;
    }
}

File: evaluation/microservice-mongodb-mongodbentities-csharp/Model/Embedded/OrdersEWithCustomerWithNationWithRegion.cs
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Entities;

namespace MongoDBEntitiesMicroservice.Model.Embedded 
{
    public class OrdersEWithCustomerWithNationWithRegion : IEntity
    {
        [BsonId]
        public int o_orderkey { get; set; }
        public Date o_orderdate { get; set; }
        public CustomerEOnlyCNameCNation o_customer { get; set; }

        public OrdersEWithCustomerWithNationWithRegion() { }

        public OrdersEWithCustomerWithNationWithRegion(int o_orderkey, Date o_orderdate, CustomerEOnlyCNameCNation? o_customer)
        {
            this.o_orderkey = o_orderkey;
            this.o_orderdate = o_orderdate;
            this.o_customer = o_customer;
        }

        public object GenerateNewID() => throw new NotImplementedException();
        public bool HasDefaultID() => false;
    }
}
